v2/dns.go

```diff
@@ -0,0 +1,143 @@
+package tsclient
+
+import (
+	"context"
+	"net/http"
+)
+
+type DNSResource struct {
+	*Client
+}
+
+type (
+	// SplitDnsRequest is a map from domain names to a list of nameservers.
+	SplitDnsRequest map[string][]string
+
+	// SplitDnsResponse is a map from domain names to a list of nameservers.
+	SplitDnsResponse SplitDnsRequest
+
+	DNSPreferences struct {
+		MagicDNS bool `json:"magicDNS"`
+	}
+)
+
+// SetSearchPaths replaces the list of search paths with the list supplied by the user and returns an error otherwise.
+func (dr *DNSResource) SetSearchPaths(ctx context.Context, searchPaths []string) error {
+	req, err := dr.buildRequest(ctx, http.MethodPost, dr.buildTailnetURL("dns", "searchpaths"), requestBody(map[string][]string{
+		"searchPaths": searchPaths,
+	}))
+	if err != nil {
+		return err
+	}
+
+	return dr.do(req, nil)
+}
+
+// SearchPaths retrieves the list of search paths that is currently set for the given tailnet.
+func (dr *DNSResource) SearchPaths(ctx context.Context) ([]string, error) {
```

Comment:
Have noticed this elsewhere, but to be explicit: we're going with the pattern of the GET operation being just the name of the thing we are getting when it doesn't interact with a top level object (e.g., top level like a `Devices().List()`) vs prepending the operation to have something like `ListSearchPaths`?

Reply:
Put differently, there is no url like `dns/searchpaths/{id}`.

Reply:
And one other way to put it - `searchPaths` behaves like a property of the `DNS` resource, as opposed to a collection of its own.

Reply:
Ah gotcha, treating it as a property on the `DNS` resource makes sense to me. I don't have super strong opinions myself but was trying to make sense of our pattern here in my head so thank you for walking through the thinking!